src/main/java/teammates/ui/controller/InstructorCoursesPageData.java

```diff
@@ -1,35 +1,192 @@
 package teammates.ui.controller;
 
-import java.util.HashMap;
+import java.util.ArrayList;
 import java.util.List;
 
 import teammates.common.datatransfer.AccountAttributes;
 import teammates.common.datatransfer.InstructorAttributes;
 import teammates.common.datatransfer.CourseAttributes;
 import teammates.common.datatransfer.CourseDetailsBundle;
+import teammates.common.util.Const;
+import teammates.common.util.Sanitizer;
+import teammates.logic.api.Logic;
+import teammates.ui.template.ActiveCoursesTable;
+import teammates.ui.template.ActiveCoursesTableRow;
+import teammates.ui.template.ArchivedCoursesTable;
+import teammates.ui.template.ArchivedCoursesTableRow;
+import teammates.ui.template.ElementTag;
 
 /**
  * This is the PageData object for the 'Courses' page 
  */
 public class InstructorCoursesPageData extends PageData {
+    private ArchivedCoursesTable archivedCourses;
+    private ActiveCoursesTable activeCourses;
+    private String courseIdToShow;
+    private String courseNameToShow;
     
     public InstructorCoursesPageData(AccountAttributes account) {
         super(account);
     }
     
-    /** Used when adding a course. Null if not adding a course. */
-    public CourseAttributes newCourse;
+    public void init(ArrayList<CourseDetailsBundle> allCourses){
+        init(allCourses, "", ""); 
+    }
+    
+    public void init(ArrayList<CourseDetailsBundle> allCourses, String courseIdToShowParam, String courseNameToShowParam){
+        activeCourses = extractActiveCourses(allCourses);
```

Comment:
It may become different from the previous version, but would changing `ArrayList` to `List` cause any issue here?